go/api/base_client.go

```diff
@@ -162,3 +163,54 @@ func (client *baseClient) Get(key string) (string, error) {
 	}
 	return handleStringOrNullResponse(result), nil
 }
+
+func (client *baseClient) Incr(key string) (int64, error) {
+	result, err := client.executeCommand(C.Incr, []string{key})
+	if err != nil {
+		return 0, err
+	}
+	return handleLongResponse(result), nil
+}
+
+func (client *baseClient) IncrBy(key string, amount int64) (int64, error) {
+	result, err := client.executeCommand(C.IncrBy, []string{key, intToString(amount)})
+	if err != nil {
+		return 0, err
+	}
+	return handleLongResponse(result), nil
+}
+
+func (client *baseClient) IncrByFloat(key string, amount float64) (float64, error) {
+	result, err := client.executeCommand(
+		C.IncrByFloat,
+		[]string{key, floatToString(amount)},
+	)
+	if err != nil {
+		return 0, err
+	}
+	return handleDoubleResponse(result), nil
+}
+
+func (client *baseClient) Decr(key string) (int64, error) {
+	result, err := client.executeCommand(C.Decr, []string{key})
+	if err != nil {
+		return 0, err
+	}
+	return handleLongResponse(result), nil
+}
+
+func (client *baseClient) DecrBy(key string, amount int64) (int64, error) {
+	result, err := client.executeCommand(C.DecrBy, []string{key, intToString(amount)})
+	if err != nil {
+		return 0, err
+	}
+	return handleLongResponse(result), nil
+}
+
+func intToString(value int64) string {
```

Comment:
Do we want to move these functions into a separate module so we could use them elsewhere?

Reply:
Done